lib/Wikia/src/Domain/User/Preferences/UserPreferences.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Wikia\Domain\User\Preferences;
+
+class UserPreferences {
+
+	/** @var GlobalPreference[] */
+	private $globalPreferences;
+
+	/** @var array[wikiId => [name => LocalPreference]] */
+	private $localPreferences;
+
+	function __construct() {
+		$this->globalPreferences = [];
+		$this->localPreferences = [];
+	}
+
+	public function setGlobalPreference($name, $value) {
+		$this->globalPreferences[$name] = new GlobalPreference($name, $value);
+		return $this;
+	}
+
+	public function setLocalPreference($name, $wikiId, $value) {
+		$this->localPreferences[$wikiId][$name] = new LocalPreference($name, $value, $wikiId);
```

Comment:
I don't know if we should impose that restriction at the service layer. It's up to whatever is using the service to make the decision to load data for multiple wikis, and if it's loading data for 500k wikis, that's a problem they should fix on their end, but we shouldn't restrict it here imo